File: gizle.cs
using System;
using System.IO;

class Program
{
    static void DosyalariBirlestir()
    {
        Console.Write("Resim dosyasının adını girin: ");
        string resimDosyasi = Console.ReadLine();

        Console.Write("RAR dosyasının adını girin: ");
        string rarDosyasi = Console.ReadLine();

        Console.Write("Oluşturulacak dosya adı: ");
        string ciktiDosyasi = Console.ReadLine();

        if (!File.Exists(resimDosyasi))
        {
            Console.WriteLine($"HATA: {resimDosyasi} bulunamadı!");
            return;
        }

        if (!File.Exists(rarDosyasi))
        {
            Console.WriteLine($"HATA: {rarDosyasi} bulunamadı!");
            return;
        }

        using (FileStream fResim = new FileStream(resimDosyasi, FileMode.Open))
        using (FileStream fRar = new FileStream(rarDosyasi, FileMode.Open))
        using (FileStream fCikti = new FileStream(ciktiDosyasi, FileMode.Create))
        {
            fResim.CopyTo(fCikti);
            fRar.CopyTo(fCikti);
        }

        Console.WriteLine("Tamamlandı: " + ciktiDosyasi);
    }

    static void Main()
    {
        DosyalariBirlestir();
    }
}

File: gizle.py
import os

def dosyalari_birlestir():
    resim_dosyasi = input("Resim dosyasının adını girin: ")
    rar_dosyasi = input("RAR dosyasının adını girin: ")
    cikti_dosyasi = input("Oluşturulacak dosya adı: ")

    if not os.path.exists(resim_dosyasi):
        print(f"HATA: {resim_dosyasi} bulunamadı!")
        return

    if not os.path.exists(rar_dosyasi):
        print(f"HATA: {rar_dosyasi} bulunamadı!")
        return

    with open(resim_dosyasi, "rb") as resim, open(rar_dosyasi, "rb") as rar, open(cikti_dosyasi, "wb") as cikti:
        cikti.write(resim.read())
        cikti.write(rar.read())

    print("Tamamlandı:", cikti_dosyasi)

dosyalari_birlestir()
